src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -82269,5 +82269,91 @@ public static void Test(bool b, object? o, G<string?> g)
                 //         G<string> g1 = g; // 5
                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInAssignment, "g").WithArguments("G<string?>", "G<string>").WithLocation(17, 24));
         }
+
+        [Fact, WorkItem(33446, "https://github.com/dotnet/roslyn/issues/33446")]
```

Comment:
>Fact [](start = 9, length = 4)

I'm trying to think of scenarios that could set a variable and are probably worth testing (if not covered already):
- `ref` and `out` arguments in `finally`, 
- deconstruction (assignment), 
- unary operator, 
- assignment to captured variable from calling a local function (I don't think that scenario is implemented yet).

Reply:
All of those use `TrackNullableStateForAssignment` which flows through the new code path.  We can add tests but they won't test any new code paths.

---
In reply to: [263935222](https://github.com/dotnet/roslyn/pull/33962#discussion_r263935222) [](ancestors = 263935222)